apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -2803,6 +2947,81 @@ static void mapHcpcs(
             });
   }
 
+  public static CareTeamComponent addCareTeamPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      ClaimCareteamrole role,
+      Optional<String> npiValue,
+      Optional<String> upinValue,
+      Optional<Boolean> includeTaxNumbers,
+      String practionerIdNumber,
+      String taxValue) {
+
+    List<Identifier> identifiers = new ArrayList<Identifier>();
+
+    if (npiValue.isPresent()) {
+      identifiers.add(
+          TransformerUtils.createPractionerIdentifier(IdentifierType.NPI, npiValue.get()));
+    }
+
+    if (upinValue.isPresent()) {
+      identifiers.add(
+          TransformerUtils.createPractionerIdentifier(IdentifierType.UPIN, upinValue.get()));
+    }
+
+    if (includeTaxNumbers.orElse(false)) {
+      identifiers.add(TransformerUtils.createPractionerIdentifier(IdentifierType.TAX, taxValue));
+    }
+
+    return addCareTeamPractitionerForPerforming(
+        eob, eobItem, role, identifiers, practionerIdNumber);
+  }
+
+  public static CareTeamComponent addCareTeamPractitionerForPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      ClaimCareteamrole role,
+      List<Identifier> identifiers,
+      String practitionerIdNumber) {
+    // Try to find a matching pre-existing entry.
+    CareTeamComponent careTeamEntry = eob.addCareTeam();
+    // addItem adds and returns, so we want size() not size() + 1 here
+    careTeamEntry.setSequence(eob.getCareTeam().size());
+
+    String practitionerId = "#practitioner-" + practitionerIdNumber;
+
+    Practitioner practitioner = findOrCreateContainedPractioner(eob, practitionerId);
```

Comment:
This "find or create" isn't going to do what you expect it to. Because `practitionerId` has the `careTeam.sequence` value embedded in it, this will **always** create a new `Practitioner`; it will never find an existing one.

In a perfect world, we would always try to reuse stuff like this where possible, to cut down on EOB sizes. But we really don't live in such a world and I'm personally inclined to say that, "it's very difficult to determine when to reuse `Practitioner` entries like this, and so we should go with the safe option and always create a new one."

Accordingly, I'd just rename and refactor the method to not try to "find" and just "create".